File: demo/ChengFenStore/Data/UserRepository.cs
using ChengFenStore.Models;
using System.Collections.Generic;
using System.Linq;
using Microsoft.EntityFrameworkCore;

namespace ChengFenStore.Data
{
    public interface IUserRepository
    {
        void AddUser(User user);
        User GetUserByPhoneNumber(string phoneNumber);
    }

    public class UserRepository : IUserRepository
    {
        private readonly AppDbContext _context;

        public UserRepository(AppDbContext context)
        {
            _context = context;
        }

        public void AddUser(User user)
        {
            _context.Users.Add(user);
            _context.SaveChanges();
        }

        public User GetUserByPhoneNumber(string phoneNumber)
        {
            return _context.Users.FirstOrDefault(u => u.PhoneNumber == phoneNumber);
        }
    }
}

File: demo/ChengFenStore/data/user_repository.py
class UserRepository:
    def __init__(self, db):
        self.db = db

    def add_user(self, user):
        self.db.session.add(user)
        self.db.session.commit()

    def get_user_by_phone_number(self, phone_number):
        return self.db.session.query(User).filter_by(phone_number=phone_number).first()
